reactor-core/src/withMicrometerTest/java/reactor/core/publisher/ContextPropagationWithScopesTest.java

```diff
@@ -0,0 +1,252 @@
+		}
+
+		assertThat(ScopeHolder.currentValue()).isEqualTo(null);
+	}
+
+	@Test
+	void subscribeFluxElsewhere() {
+		AtomicReference<ScopedValue> valueInNewThread = new AtomicReference<>();
+
+		ScopedValue externalValue = ScopedValue.create("outside");
+		ScopedValue internalValue = ScopedValue.create("inside");
+
+		try (ScopedValue.Scope scope = externalValue.open()) {
+			assertThat(ScopeHolder.currentValue()).isEqualTo(externalValue);
+
+			Flux.just(1)
+			    .subscribeOn(Schedulers.single())
+			    .doOnNext(i -> {
+				    valueInNewThread.set(ScopeHolder.currentValue());
+			    })
+			    .contextWrite(Context.of(ScopedValueThreadLocalAccessor.KEY, internalValue))
+			    .blockLast();
+
+			assertThat(valueInNewThread.get()).isEqualTo(internalValue);
+			assertThat(ScopeHolder.currentValue()).isEqualTo(externalValue);
+		}
+
+		assertThat(ScopeHolder.currentValue()).isEqualTo(null);
+	}
+
+	@Test
+	void multiLevelScopesWithDifferentValuesAndFlux() {
+		ScopedValue v1 = ScopedValue.create("val1");
+		ScopedValue v2 = ScopedValue.create("val2");
+
+		AtomicReference<ScopedValue> valueInsideFlatMap = new AtomicReference<>();
+
+		try (ScopedValue.Scope v1scope1 = v1.open()) {
+			try (ScopedValue.Scope v1scope2 = v1.open()) {
+				try (ScopedValue.Scope v2scope1 = v2.open()) {
+					try (ScopedValue.Scope v2scope2 = v2.open()) {
+						try (ScopedValue.Scope v1scope3 = v1.open()) {
+							try (ScopedValue.Scope nullScope =
+									     ScopedValue.nullValue().open()) {
+								assertThat(ScopeHolder.currentValue().get()).isNull();
+							}
+							assertThat(ScopeHolder.currentValue()).isEqualTo(v1);
+							assertThat(ScopeHolder.get()).isEqualTo(v1scope3);
+
+							Flux.just(1)
+							    .flatMap(i ->
+									    Flux.just(i)
+									        .publishOn(Schedulers.boundedElastic())
+									        .doOnNext(item -> valueInsideFlatMap.set(ScopeHolder.currentValue())))
+							    .blockLast();
+
+							assertThat(ScopeHolder.currentValue()).isEqualTo(v1);
+							assertThat(ScopeHolder.get()).isEqualTo(v1scope3);
+						}
+						assertThat(ScopeHolder.currentValue()).isEqualTo(v2);
+						assertThat(ScopeHolder.get()).isEqualTo(v2scope2);
+					}
+					assertThat(ScopeHolder.currentValue()).isEqualTo(v2);
+					assertThat(ScopeHolder.get()).isEqualTo(v2scope1);
+				}
+				assertThat(ScopeHolder.currentValue()).isEqualTo(v1);
+				assertThat(ScopeHolder.get()).isEqualTo(v1scope2);
+			}
+			assertThat(ScopeHolder.currentValue()).isEqualTo(v1);
+			assertThat(ScopeHolder.get()).isEqualTo(v1scope1);
+		}
+
+		assertThat(ScopeHolder.currentValue()).isNull();
+
+		assertThat(valueInsideFlatMap.get()).isEqualTo(v1);
+	}
+
+	@Test
+	void multiLevelScopesWithDifferentValuesAndMono() {
+		ScopedValue v1 = ScopedValue.create("val1");
+		ScopedValue v2 = ScopedValue.create("val2");
+
+		AtomicReference<ScopedValue> valueInsideFlatMap = new AtomicReference<>();
+
+		try (ScopedValue.Scope v1scope1 = v1.open()) {
```

Comment:
in what sense do you mean? modifying the v1scope1 or opening a new scope?

Reply:
I was thinking about 2 cases: 

 - one call scope close() manually (probably that would break everything and should be illegal 🧐)
 - one scope open and then close manually somewhere else up the stack  e.g.
 		```java
          var scope = ...
 		   mono
                   .doOnSubscibe(__-> scope.set(Scope.open()))
                   .doAfterTerminate(__-> scope.get().close())
         ```          
         which should be probably legal and should not break anything (but I'm not sure)

Reply:
> one call scope close() manually (probably that would break everything and should be illegal 🧐)

Yes, this is illegal and the behaviour should be undefined. Adding tests for it would become some sort of a specification.

> one opens a scope manually and then closes it manually somewhere else in the chain e.g.
java var scope = ... mono .doOnSubscibe(__-> scope.set(Scope.open())) .doAfterTerminate(__-> scope.get().close()) 
which should be probably legal and should not break anything (but I'm not sure)

In this case it's also illegal in my opinion - the reactive chain is agnostic of Thread switches. Scopes are Thread-bound. If you open a Scope in one Thread, you must close it in the same Thread. Doing that using reactive operators is dangerous, I'd avoid adding a test that shows such a pattern.